error.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+// We define a custom error type so that we can provide friendlier error messages
+type fetchError struct {
+	errorCode int    // an error code is an arbitrary int that allows for strongly typed identification of specific errors
+	details   string // the output of the underlying error message, if any
+	error     error // the underlying golang error, if any
```

Comment:
The field name probably shouldn't be `error`, as that conflicts with the type of the same name. Most go code I've seen calls it `err`. `underlyingErr` would also work.